### DISCUSSION TITLE
Shift heatmap background half a cell

### DISCUSSION THREAD

**Discussion by mmatous** [mwaskom/seaborn#3480] [Q&A]
Is there a way to shift heatmap background (missing values depiction) in such a way that the background cells match alignment of the plotted cells? I tried `linewidths=1` but while it makes them look aligned in reality it splits the background cells so they are composed of 4 smaller cells.
Or maybe it's possible to overlay heatmaps where the background one would plot with monocolored palette? I couldn't figure out how to create that.

**Comment by thuiop** [comment]
An example would help a lot;

**Comment by mmatous** [comment]
It's just a regular invocation, no bells or whistles.
```
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

df = pd.DataFrame([[1, 2, 3], [4, 5, None], [None, None, None]], columns=[1, 2, 3])
sns.set_theme()
sns.heatmap(df)
plt.show()
```

**Comment by mwaskom** [comment]
It's a little unclear what you mean by "background cells" but I think you're talking about the axes grid that shows through where there is missing data? You can change you code to have

```python
ax = sns.heatmap(df)
ax.grid(False)
```
![image](https://github.com/mwaskom/seaborn/assets/315810/8e37aa46-2ad4-44c3-b162-4691c6b73b92)

PS even without bells or whistles it's important to provide an example because (1) other people can't see what you're looking at and the description wasn't really clear and (2) you're asking for people to help you and it will be a lot easier if you give them something to start with.

**Comment by mmatous** [comment]
The grid lines sort of make up cells, similar to cells in a heatmap, just a bit shifted. I know they can be toggled off completely but this is what I'm trying to achieve:
![test](https://github.com/mwaskom/seaborn/assets/18654729/c80a936a-d4c4-437e-bd4f-6eda3a818c0b)

If seaborn can't do it, I'm pretty sure I'm supposed to use [matplotlib.transforms.Affine2D.translate](https://matplotlib.org/stable/api/transformations.html#matplotlib.transforms.Affine2D.translate). But how do I figure out how many pixels do I need?

**Comment by mwaskom** [comment]
Se `linewidth` in heatmap and turn off the background grid?

**Comment by mmatous** [comment]
Right. I feel dumb now. :facepalm:  I didn't originally plan on having the lines in the actual heatmap, just the grey background, and thus got hyperfocused on the shift/transform thing, but this is good enough. More importantly, it's nice and straightforward. Thank you :)
